app/models/sipity.rb

```diff
@@ -105,4 +105,9 @@ def handle_conversion(input, result, method_name)
     raise ConversionError.new(input) # rubocop:disable Style/RaiseArgs
   end
   module_function :handle_conversion
+
+  def hyrax_or_valkyrie_global_id(input)
+    Entity.find_by(proxy_for_global_id: Hyrax::GlobalID(input).to_s).nil? ? Hyrax::GlobalID(ActiveFedora::Base.find(input.id.id)) : Hyrax::GlobalID(input)
```

Comment:
what's going on here?

i think the `ActiveFedora::Base.find(input.id.id)` won't be okay. this is a whole database round trip to get (what looks like the wrong) `GlobalID`

Reply:
Yeah, this is just the remedy for the problem. Im sure you all would like to really discuss how you would like to truly fix this problem. We can chat at standup if needed. This is a systemic problem with how the AF::Base and Valkyrie mechanism works when transforming a AF::Base object to a Valkyrie object. 

Basically, if an AF::Base object exists in the DB and gets transformed to a Valkyrie object by way of `Hyrax.query_service.find_by_alternate_identifier(alternate_identifier: params[:id], use_valkyrie: true)`, the Valkyrie object doesn't have a Sipity::Entity that exists in the DB for the Valkyrie object. So when trying to process the workflow, it fails because the GlobalID for the Valkyrie object isn't the same as the AF::Base Global::ID